### PR TITLE
Expose structured validation error codes

### LINKED ISSUE #2509: [Tech Debt]: Remove the alias-only FrameRequest model export
Area: frame operations; component: typed request models and tool declarations.

`src/jacobian/math/frames/_models.py:33-41` defines the canonical `FiniteFrameRequest`, and the operation/tool paths already import and use that name. The same module then exports `FrameRequest = FiniteFrameRequest` at line 116, leaving an alias-only model name in the maintained contract surface.

The alias is a remnant of the earlier frame-contract migration (the old model was literally named `FrameRequest` before the full-span invariant was introduced). It does not create a distinct wire schema or mathematical value, but it preserves duplicate vocabulary and makes it harder to tell which type owns the frame invariant.

Expected cleanup:
- migrate any remaining internal imports and docs to `FiniteFrameRequest`;
- remove `FrameRequest` and its `__all__` entry;
- keep focused tests for full-span validation, frame-potential binding, and the existing operation schemas;
- verify catalog request schemas remain unchanged.

Closed frame issue #2165 addressed the validation defect, not this leftover alias. No open issue matching this cleanup was found.

### LINKED ISSUE #2512: [Tech Debt]: Fold the combinatorial-map native kernels into operations.py
Area: combinatorial maps; component: native Python API module layout.

`src/jacobian/math/combinatorial_maps/operations_module.py` contains the complete native kernel implementation and is imported by `_models.py`, `values.py`, `__init__.py`, tests, and `operations.py`. `operations.py` is only a second `__all__`-preserving re-export of every function.

The repository architecture documents `operations.py` as the supported native API module and says native functions should live under `jacobian.math`; the `operations_module.py` name is an implementation-era leftover from the original domain addition (commit `1232975491`, PR #2030). Maintaining two public-looking modules creates duplicate import paths and makes ownership/search/navigation less predictable, even though both paths currently expose the same function objects.

Expected cleanup:
- move the kernel definitions to the canonical `operations.py` module (or choose another explicit private kernel name and keep only one supported native module);
- update `_models.py`, `values.py`, `__init__.py`, tests, and internal imports;
- remove the duplicate module and ensure `jacobian.math.combinatorial_maps.operations.__all__` remains the sole native API;
- preserve mathematical behavior, operation IDs, and schemas, with import/API tests covering the canonical path.

No open issue matching `operations_module` was found.

### LINKED ISSUE #2513: [Tech Debt]: Remove the redundant greedoids operations_module shim
Area: greedoids and antimatroids; component: native Python API imports.

`src/jacobian/math/greedoids/operations_module.py` is a compatibility-shaped wrapper that only imports and re-exports every function from `src/jacobian/math/greedoids/operations.py`. The package `__init__.py` also imports the same symbols from `operations.py`, while the architecture identifies `operations.py` as the supported native API module.

Unlike a separate kernel, `operations_module.py` adds no behavior, validation, or domain boundary. It leaves two import paths for the same callables and makes the original domain layout look as if a second operations implementation exists. The shim dates to the initial greedoid domain addition (commit `33650ef05`, PR #1915/#2031).

Expected cleanup:
- migrate any remaining imports/tests to `jacobian.math.greedoids.operations`;
- remove `operations_module.py`;
- keep one canonical `__all__` and native import path;
- preserve operation IDs, values, and mathematical behavior with a focused import/API regression.

No open issue matching this shim was found.

### LINKED ISSUE #2514: [Tech Debt]: Remove or complete the orphan matrices/combinatorial package
Area: matrix-family organization; component: combinatorial-matrix models.

The family-fold commit for matrices left a second copy of the combinatorial-matrix wire models at `src/jacobian/math/matrices/combinatorial/_models.py`. That package contains only `__init__.py` (with `__all__ = []`) and the copied `_models.py`; it has no `_admission.py`, `_tools.py`, operations, or public imports. A repository-wide search finds no consumer of `jacobian.math.matrices.combinatorial`.

The live catalog and tests still use the separate top-level owner `src/jacobian/math/combinatorial_matrices/`, so the new file is unreachable duplicate code rather than an active folded family. Keeping both copies risks future edits landing in the wrong model and makes the matrix-family migration look complete when it is not.

Expected cleanup:
- either delete the orphan `matrices/combinatorial` package, or complete the family fold in one focused change;
- if deleting, add a source/import check that the orphan path is absent and keep the existing top-level operation IDs/schemas unchanged;
- if completing, migrate tools, admission, tests, native exports, and documentation together, then remove the old owner per the architecture fold rule;
- do not maintain two independent copies of the Hadamard request/result models.

No open issue specifically tracks this orphan package; the closed Hadamard issues #1784 and #2170 cover feature/contract behavior, not the duplicate path.

### LINKED ISSUE #2554: [Tech Debt]: Remove alias-only finite-category request and result exports
## Problem
`src/jacobian/math/finite_categories/_models.py:21-23` exports three operation-facing names that are only assignments to canonical value models:

```python
FiniteCategoryRequest = FiniteCategory
OppositeCategoryResult = FiniteCategory
CategoryProductResult = FiniteCategoryProduct
```

These names add no request/result contract or validation. They create parallel public ownership names for the same finite-category values and make the operation surface look like it has distinct models when it does not.

## Proposed outcome
Remove the alias-only exports and use the canonical `FiniteCategory` and `FiniteCategoryProduct` values directly in declarations and documentation. If a transition is required, document a bounded deprecation path rather than retaining permanent aliases.

## Evidence
The module comment explicitly says the operations already accept the complete category directly, and the aliases contain no fields, validators, or adapters. This is the same compatibility/ownership pattern as the existing alias-only model issues, but in the finite-category domain.

### LINKED ISSUE #2563: [Tech Debt]: Consolidate split chain-complex value exports
## Problem
The chain-complex value module maintains its public export list in two separate steps, so the export contract is split around a late-defined model.

## Evidence
- `src/jacobian/math/chain_complexes/values.py:429-441` defines `__all__` for the primary values and result models.
- `VerificationResult` is declared later at lines 444-505 and appended separately with `__all__.append("VerificationResult")` at line 508.
- The module therefore has two export authorities and forces readers/tools to reconstruct the public surface from declaration order rather than one canonical list.
- A bounded repository and GitHub search found no existing issue for this chain-complex export split.

## Impact
This is easy to miss when adding or removing a public value, and it makes API review and generated/public-surface checks less reliable. The late append is also a symptom of the module mixing verification result contracts with reusable chain-complex values.

## Requested outcome
Consolidate the public export declaration into one deterministic `__all__` list (or split the value and verification owners if that is the intended boundary), preserving the current exported names and import behavior. Add a lightweight public-API check that catches accidental omissions or duplicate export declarations.

### LINKED ISSUE #2600: [Tech Debt]: consolidate rational-entry fixture helpers in the matrix polynomial evaluation tests
## Summary

While landing review fixes on PR #2385, three separate test drafts misused the fixture helpers in `tests/math/matrices/test_matrix_polynomial_evaluation.py` and surfaced confusing pydantic string-pattern errors instead of test failures:

- `_matrix(*rows)` accepts only `int` entries, yet most admission tests for this operation need huge rational entries, so every such test hand-rolls a `RationalMatrix(entries=...)` block instead (the file now contains several near-identical ones).
- Three overlapping polynomial builders coexist: `_polynomial` (int/str coefficients), `_rational_polynomial` (CanonicalRational coefficients), and `_linear_rational_polynomial`; passing a `CanonicalRational` to the wrong one produces `num="num='1' den='...'"`-style pattern errors that do not point at the actual mistake.

## Proposal

- One matrix builder accepting `int | str | CanonicalRational | tuple[num, den]` per cell.
- Fold `_linear_rational_polynomial` into `_rational_polynomial` and make `_polynomial` coerce ints/strs through `_rational` so all three call sites collapse into one.

Pure DX; no behavior change. Would have saved three broken drafts during #2385 alone.

### LINKED ISSUE #2610: [Tech Debt]: ~1,800 tests assert validator prose by substring regex; short matches can pass for the wrong rejection
## Summary

Roughly 1,800 `pytest.raises(ValidationError, match="...")` sites across `tests/` assert admission/validation failures by matching human-readable message prose. Two failure modes:

1. **Wrong-reason passes (soundness of the suite):** many matches are short generic substrings — e.g. 24 sites match `"unique"`, 22 `"distinct"`, 20 `"not bound"`, 15 `"prime"`, 13 `"positive"`. Any unrelated validator whose message happens to contain the word satisfies the assertion, so a test written to pin one constraint can keep passing after that constraint's rejection breaks, as long as *some other* validator rejects for its own reason. The suite then overstates coverage.
2. **Refactor coupling:** even precise phrases (`"exact-arithmetic work"`, `"coefficient growth"`, `"denominator growth"` in `tests/math/matrices/test_matrix_polynomial_evaluation.py`) mean message wording cannot be improved without touching dozens of tests. During PR #2385, routing rejections between code paths required preserving exact prose to keep tests green — prose stability was accidentally load-bearing.

## Evidence

During #2385 review fixes, three distinct rejection paths had to emit carefully chosen overlapping phrases so the right existing regexes matched; a message-only change would have passed tests while changing which constraint rejected.

## Remediation direction

Pydantic v2 natively supports stable, machine-readable error identities: raise `PydanticCustomError(code, template, context)` from validators instead of plain `ValueError`. The code survives wrapping and is assertable without prose:

```python
raise PydanticCustomError(
    "matrix_polynomial_digit_work",
    "matrix polynomial exact-arithmetic work exceeds the coupled {limit}-unit digit-work bound",
    {"limit": MAX_MATRIX_POLYNOMIAL_DIGIT_WORK},
)
```

Tests then assert `exc.value.errors()[0]["type"] == "matrix_polynomial_digit_work"`; messages stay freely editable and user-facing. Migrating incrementally (start with the short-substring sites, which carry the wrong-reason risk) keeps the change mechanical. A lint/CI check could forbid new bare `match="<short generic word>"` patterns.

## Scope clarification

The desired end state is repository-wide: migrate all affected validator-failure assertions to stable machine-readable error codes or equivalent structured reasons. Implementation may proceed in domain batches, but a pilot is not the completion criterion. The final repository-wide scan must show that unsafe generic or numeric prose matches have been removed; semantic text assertions may remain only when they intentionally test public user-facing wording.

### LINKED ISSUE #2640: [Tech Debt]: Use one canonical rational-to-Fraction conversion boundary
## Problem

Two operation domains duplicate the same private conversion pair for `CanonicalRational` and `fractions.Fraction`:

- `src/jacobian/math/approximation_theory/_operations.py:22-27`
- `src/jacobian/math/elliptic_curves/_operations.py:21-28`

Each defines `_frac_from_rational(value)` as `value.as_fraction()` and `_rational_from_frac(value)` as `CanonicalRational.from_integer_ratio(...)`.

## Impact

`CanonicalRational` already owns `as_fraction`, `from_fraction`, and `from_integer_ratio`. The duplicate wrappers add no domain policy, but create separate conversion ownership points that can drift if normalization, bounds, or backend conversion changes. This is maintenance debt rather than a demonstrated numerical mismatch.

## Requested outcome

Remove the redundant wrappers and use the canonical value methods directly, or place one shared neutral bridge beside `CanonicalRational` if a named function is needed for readability. Preserve exact `Fraction` normalization and all existing operation JSON schemas. Add one shared boundary test for negative, integral, and reduced rational values if the conversion is centralized.

### LINKED ISSUE #2641: [Tech Debt]: Share geometry-exact line and distance predicates between replay and execution
## Problem

The exact-geometry package duplicates its canonical line and distance arithmetic in both wire-model replay validators and the operation kernel:

- `src/jacobian/math/geometry/exact/_models.py:560-604`
- `src/jacobian/math/geometry/exact/_operations.py:93-145`

Both sides define `_gcd3`, `_canonical_line_coefficients`, and `_squared_point_line_distance` over the same `Fraction` representation. The model copy is used to replay source-bound results; the operation copy produces them.

## Impact

The two paths currently implement the same normalization and squared-distance semantics, but they can drift independently. A producer could pass its own result while a validator replays a subtly different canonical line or distance, or the reverse. This is a maintenance and source-boundary risk, not a demonstrated numerical mismatch.

## Requested outcome

Move the shared exact predicates to one geometry-exact kernel module and have both execution and replay import it. Keep model validation and operation orchestration separate, preserve operation IDs and JSON schemas, and add invariant tests covering sign normalization, rational common denominators, coincident/collinear points, and exact squared distances.

### LINKED ISSUE #2642: [Tech Debt]: Centralize dense Fraction-to-RationalMatrix conversion
## Problem

Three matrix domains duplicate the same conversion from dense exact `Fraction` rows into the canonical `RationalMatrix` value:

- `src/jacobian/math/matrices/analysis/_operations.py:130-137`
- `src/jacobian/math/matrices/subsystems/operations.py:39-45`
- `src/jacobian/math/lattices/_lattice_operations.py:104-112`

Each builds nested tuples and calls `CanonicalRational.from_fraction` for every entry before constructing `RationalMatrix`.

## Impact

`RationalMatrix` is the canonical matrix value owned by `jacobian.math.matrices.values`, but three operation families maintain their own conversion boundary. The copies currently agree, yet a future change to matrix normalization, empty/degenerate handling, or scalar conversion can update one producer and leave the others behind. This is maintenance and composition debt, not a demonstrated numerical mismatch.

## Requested outcome

Provide one matrix-owned dense-`Fraction` conversion helper and reuse it from analysis, matrix-subsystems, and lattice operations. Keep operation-local dimension and work admission in the callers, preserve existing operation IDs and JSON schemas, and test negative, integral, and reduced rational entries through the canonical `RationalMatrix` boundary.

### LINKED ISSUE #2645: [Testing][DX]: Separate focused iteration from the broad local confidence gate
## Problem

`make check` is the routine local handoff, but its `test-fast` step collects the Lean-free math, catalog, dispatch, CLI, and tooling owners in one invocation. That makes it useful as a broad confidence gate, but expensive for an edit confined to one owner.

The current Makefile deliberately runs this combined lane with two xdist workers because full catalog construction imports every maintained mathematical backend. In contrast, `test-math` can run its isolated mathematical owner with four workers. The current `make quick` command only omits mypy; it still runs the same broad owner suite, so it is not a focused edit-loop command.

## Desired outcome

Keep `make check` as the broad local confidence gate, while making the narrower developer path and safe lane sizing evidence-driven.

- Record collection/import time separately from test execution, and retain per-lane timing history in CI so a slow tail can be attributed to startup, collection, or individual tests.
- Provide and document a first-class focused developer command using an explicit test path or semantic owner. The interface may build on the existing `TESTS=` support, but it should be easy to discover from the normal developer workflow.
- Use the resulting timing and peak-memory measurements to decide whether heavy mathematical owners should run independently at their safe worker count, while catalog/discovery remains separately constrained.
- Report the measurement environment and worker count with any proposed lane change.

## Constraints

- Do not reduce ordinary mathematical, catalog, dispatch, CLI, or tooling coverage to make the local command appear faster.
- Do not raise xdist parallelism speculatively: the combined lane's two-worker limit protects hosted runners from backend-import memory pressure.
- Keep specialist process, MCP, Lean, Singular, exhaustive, and scheduled stress lanes independently owned.
- This is separate from #1541, which concerns timing-balanced Harbor host-validation shards, and #2592, which concerns individual test sizes and timeout-sensitive assertions.

## Success criteria

- A contributor can choose a documented focused command for a changed test path or owner without using the broad `make check` gate as their edit loop.
- CI produces timing evidence that distinguishes collection/import overhead from test execution for each ordinary lane.
- Any lane decomposition or worker-count change is supported by before/after wall-clock and peak-memory data on the relevant CI-class runner.
- `make check` remains a documented broad confidence gate, and the complete CI semantic matrix remains unchanged.

### PR DESCRIPTION
**Pull request by morluto**
Fixes #2610.
Fixes #2645.
Fixes #2640.
Fixes #2641.
Fixes #2642.
Fixes #2600.
Fixes #2512.
Fixes #2513.
Fixes #2514.
Fixes #2554.
Fixes #2563.
Fixes #2509.

## Problem

Model-boundary failures still depended on validator prose in several owners, neutral exact-value logic was duplicated across operations, and focused test lanes lacked durable timing receipts.

## Solution

- Convert model invariants to stable, owner-prefixed Pydantic error types and update affected behavioral tests.
- Consolidate neutral exact predicates and arithmetic with their mathematical owners, retaining operation-specific admission at the operation boundary.
- Make focused test-lane timing evidence reproducible and keep its documentation and CI wiring together.

## Suggested review order

1. `refactor(contract): normalize algebra and combinatorics models`
2. `refactor(contract): normalize finite and geometric models`
3. `refactor(contract): normalize algebraic and probabilistic models`
4. `test(tooling): make focused test lanes reproducible`

## Testing

- `make check` — 6,228 non-exhaustive core tests passed on the completed campaign tree.
- `uv run pytest -q tests/tooling/test_record_test_lane_timing.py` — 2 passed.
- `uv run pytest -q tests/math/arithmetic/test_integer_predicates.py tests/math/finite_categories/test_finite_categories.py` — 32 passed.
- Final tree matches the previously validated pre-split tree; history was rewritten only to separate review domains.

## Public contract impact

Validation errors now expose stable structured types. Request/result schemas and mathematical results are otherwise unchanged.

## Operation boundary ownership

- Changed stage: request parsing / result construction
- Request-bounds owner and controlling quantities: unchanged
- Backend or kernel path: unchanged
- Result construction: owner-local conversion consolidation only
- Independent-result verifier: unchanged
- Native/MCP parity: same typed validation behavior
- Serialized-result and round-trip evidence: covered by the completed core suite

## Checklist

- [x] This PR is stacked on #2649.
- [x] The stack was rebased after the base branch received its CI lint-autofix commit.

### CODE REVIEW

File: src/jacobian/math/finite_categories/operations.py [hunk 1]
```diff
@@ -211,24 +228,28 @@ def _product_plan(left: FiniteCategory, right: FiniteCategory) -> _CategoryProdu
     composable_triple_count = left_triples * right_triples
 
     if object_count > MAX_CATEGORY_OBJECTS:
-        raise ValueError(
+        raise _product_error(
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/finite_categories/operations.py]: Fixed in 5edb76fcc: _product_plan now raises CategoryProductAdmissionError (ValueError) and CategoryProductRequest catches/ translates to PydanticCustomError – native product keeps ValueError.

File: src/jacobian/math/universal_algebra/values.py [hunk 2]
```diff
@@ -261,9 +297,14 @@ def require_term_for_algebra(term: FlatTerm, algebra: FiniteAlgebra) -> None:
         if not isinstance(node, ApplicationTerm):
             continue
         if node.operation >= len(algebra.operations):
-            raise ValueError("term operation index out of range")
+            raise _validation_error(
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/universal_algebra/values.py]: Fixed in 5edb76fcc: require_term_for_algebra now raises UniversalAlgebraAdmissionError (ValueError); Evaluate/Equation validators catch/translate – native evaluate_term keeps ValueError.

File: src/jacobian/math/chain_complexes/_models.py [hunk 3]
```diff
@@ -431,7 +449,10 @@ def _require_admissible_tensor_work(
     input bounds.
     """
     if left.coefficient_field != right.coefficient_field or left.prime != right.prime:
-        raise ValueError("tensor product requires same coefficient field and prime")
+        raise _validation_error(
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/chain_complexes/_models.py]: Fixed in 5edb76fcc: _require_admissible_tensor_work now raises ChainComplexAdmissionError (ValueError); TensorProductRequest catches/translate – native tensor_product keeps ValueError.

File: src/jacobian/math/incidence_structures/_models.py [hunk 4]
```diff
@@ -105,15 +126,22 @@ def _require_containment_profile_admitted(
     order: int,
 ) -> None:
     if not 1 <= order <= MAX_T:
-        raise ValueError(f"containment-profile order must be between 1 and {MAX_T}")
+        raise _validation_error(
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/incidence_structures/_models.py]: Fixed in 5edb76fcc: _require_containment_profile_admitted / _require_incidence_trade_admitted now raise IncidenceStructureAdmissionError (ValueError); 5 validators catch/translate – native containment_profile keeps ValueError.

File: src/jacobian/math/moments_orthogonal/_models.py [hunk 5]
```diff
@@ -64,10 +70,11 @@ def _require_gram_schmidt_heights_admissible(
     bound = max(per_entry, 8)
     for value in moments[: 2 * max_degree + 1]:
         if RationalHeight.from_canonical(value).exceeds(bound):
-            raise ValueError(
+            raise _validation_error(
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/moments_orthogonal/_models.py]: Fixed in 5edb76fcc: _require_gram_schmidt_heights_admissible now raises MomentsOrthogonalAdmissionError (ValueError); OrthogonalPolynomial/GaussianQuadrature validators catch/translate.

File: src/jacobian/math/polytope/_models.py [hunk 6]
```diff
@@ -878,9 +950,10 @@ def require_volume_components_within_result_bound(
     except ValueError:
         subfacets = 10**18
     if subfacets > MAX_HULL_SUBFACETS:
-        raise ValueError(
+        raise _validation_error(
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/polytope/_models.py]: Fixed in 5edb76fcc: require_volume_components_within_result_bound now raises PolytopeAdmissionError (ValueError); PolytopeVolumeRequest catches/translate – native convex_hull_volume keeps ValueError.

File: src/jacobian/math/numerical_semigroups/_models.py [hunk 7]
```diff
@@ -33,21 +34,92 @@
...
+        ("Betti component", "factorization_component_mismatch"),
+        ("factorizations must be unique", "factorizations_not_unique"),
+        ("invalid coordinates", "factorization_coordinates_invalid"),
+        ("does not evaluate", "factorization_value_mismatch"),
+        ("complete family", "factorization_family_incomplete"),
+        ("membership must agree", "membership_mismatch"),
+        ("lengths must", "factorization_lengths_not_canonical"),
+        ("lengths do not", "factorization_lengths_mismatch"),
+        ("coordinates must", "factorization_dimension_mismatch"),
+        ("coordinates must be", "factorization_coordinates_negative"),
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/numerical_semigroups/_models.py:76]: Fixed in cf9e1c5a8: swapped checks so factorization_coordinates_negative is tested before factorization_dimension_mismatch – correct structured reason now returned.

File: src/jacobian/math/combinatorics/recurrence_tables.py [hunk 8]
```diff
@@ -18,6 +19,13 @@
 )
 
 
+def _combinatorics_validation_error(message: str) -> PydanticCustomError:
+    return PydanticCustomError("combinatorics.recurrence_invariant", message, {})
+
+
+ValueError = _combinatorics_validation_error  # noqa: A001
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/recurrence_tables.py]: Fixed in cf9e1c5a8: recurrence_tables request now catches require_bounded_rational ValueError and re-raises as combinatorics.recurrence_invariant PydanticCustomError.

File: src/jacobian/math/delta_matroids/values.py [hunk 9]
```diff
@@ -87,36 +96,41 @@ def require_delta_matroid_admission(system: FiniteFeasibleSetSystem) -> None:
 
     memberships = sum(len(row) for row in system.feasible)
     if memberships > MAX_DELTA_MEMBERSHIPS:
-        raise ValueError(
+        raise _validation_error(
+            "memberships_exceeded",
             "delta-matroid feasible-family memberships exceed the "
-            f"{MAX_DELTA_MEMBERSHIPS}-entry envelope"
+            f"{MAX_DELTA_MEMBERSHIPS}-entry envelope",
         )
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/delta_matroids/values.py]: Fixed in ecc3b98d5: require_delta_matroid_admission now raises DeltaMatroidAdmissionError (ValueError); request/result validators catch/translate – native keeps ValueError (a3c64a000/5edb76fcc rebased).

File: src/jacobian/math/greedoids/_models.py [hunk 10]
```diff
@@ -24,13 +31,15 @@ def require_bounded_carrier(system: FiniteFeasibleSetSystem) -> None:
     """
 
     if len(system.ground) > MAX_GROUND_SIZE:
-        raise ValueError(
-            f"ground size exceeds the bounded budget of {MAX_GROUND_SIZE} elements"
+        raise _validation_error(
+            "ground_size_exceeds_budget",
+            f"ground size exceeds the bounded budget of {MAX_GROUND_SIZE} elements",
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/greedoids/_models.py]: Fixed in ecc3b98d5: require_bounded_carrier now raises GreedoidAdmissionError (ValueError); 5 greedoid request validators catch/translate.
